Marlin/src/module/temperature.h

```diff
@@ -699,7 +699,7 @@ class Temperature {
 
       // Convert the given heater_id_t to idle array index
       static IdleIndex idle_index_for_id(const int8_t heater_id) {
-        TERN_(HAS_HEATED_BED, if (heater_id == H_BED) return IDLE_INDEX_BED);
+        OPTCODE(HAS_HEATED_BED, if (heater_id == H_BED) return IDLE_INDEX_BED)
```

Comment:
FYI — `OPTCODE` was created out of an abundance of caution because compilers will bark at a stray semicolon in some situations (typically, inside of a `switch` block). If the stray semicolon at the end of a `TERN_` would not cause the compiler to bark, then `TERN_` should be preferred. This is because "dumb" C/C++ syntax parsers in most text editors and IDEs expect a semicolon at the end of a statement.

Reply:
that makes sense. actually what I thought was seeing `OPTCODE` used in a 'switch case' I looked for similar situations. and where I had seen `return` I confused that with being a 'switch case', then sorta went overkill on the other ones. I had seen all the other `TERN_` where if it were disabled it would leave a rouge `;`. I knew it wouldn't leave an issue but figured why not give it a try.

for instance, (not seriously considering) we could technically replace ALL the `TERN_` with `OPTCODE` - just saying.  
or what about moving the semicolon within the confounds of the parenthesis? (not suggesting, just speculating)  
would this be the more correct approach? but as you said it should not matter because it shouldn't cause an issue either way.

on a side note, i'll go ahead and revert back the changes here.